### ISSUE TITLE
Broken test BibTeXItemDataProviderTest.noCollectionAuthor

### ISSUE THREAD

**Issue by gallardo** [michel-kraemer/citeproc-java#66]
(PR coming soon)

**Comment by gallardo** [comment]
Reviewing this issue and after noticing that the referred test was already addressed by commit https://github.com/michel-kraemer/citeproc-java/commit/67297f8a949a2d6053a0c59561071af03cd4c58a to fix https://github.com/michel-kraemer/citeproc-java/issues/58 and match the (by then) wrong rendered entry, I'm analyzing it in more detail.

I'm a bit uncomfortable with the test data:
https://github.com/michel-kraemer/citeproc-java/blob/0b9a42fabaa543e4aff3195fbc304a475771a5e7/citeproc-java/src/test/java/de/undercouch/citeproc/bibtex/BibTeXItemDataProviderTest.java#L179-L186

The data stems apparently from an issue:

```
@book{EconBiz-10009450595,
title = {{Essays on the Role of Specific Human Capital}},
author = {Hu, Xiaohan},
editor = {Hellerstein, Judith},
year = {2007-06-04},
type= {Thesis},
language= {eng},
}
```

_Originally posted by @alinamx in https://github.com/michel-kraemer/citeproc-java/issues/38#issuecomment-326205009_

but to be precise, books don't support the field "type", and for thesis there are better bib entry types `@thesis`, `@mastersthesis` and `@phdthesis` (see [biblatex documentation, §2.1.1](http://mirrors.ctan.org/macros/latex/contrib/biblatex/doc/biblatex.pdf)). I have imported the  metadata that I could find in internet for this publication (https://drum.lib.umd.edu/handle/1903/7186) in Zotero, to export it to bibtex. This would probably be a better test data record:

```
@phdthesis{hu_xiaohan_essays_2007,
	type = {Dissertation},
	title = {Essays on the {Role} of {Specific} {Human} {Capital}},
	language = {en\_US},
	school = {University of Maryland (College Park, Md.)},
	author = {{Hu, Xiaohan}},
	collaborator = {{Hellerstein, Judith}},
	month = jun,
	year = {2007}
}
```
but this entry has the disadvantage of not helping to test the issue targeted by this method (thesis don't support the field "editor").

For testing purposes, I'd advocate the use of a book entry from the `unix.bib` collection, so that we have more confidence in the validity of the source test-data and the expectation is easier to predict. E.g.:

```
@book{rice_five_1993,
	address = {Santa Fe, NM},
	title = {Five steps to {HP}-{UX}},
	isbn = {0-934605-24-6},
	publisher = {OnWord Press},
	editor = {Rice, Jim},
	year = {1993},
	lccn = {QA76.8.H48 F58 1993}
}
```
Expected:

> Rice, J. (Ed.). (1993). Five steps to HP-UX. Santa Fe, NM: OnWord Press.

**Comment by michel-kraemer** [comment]
I'd rather leave it as it is (see my comment in the pull request). The library should be able to handle incorrect input data and the test runs successfully anyhow. So why change it?

**Comment by gallardo** [comment]
(True, the inline comment is currently not valid. I could extend the discussion on #38, to not loose the context, or I could remove the comment.)

With commit https://github.com/michel-kraemer/citeproc-java/commit/67297f8a949a2d6053a0c59561071af03cd4c58a I forced the expectation to match an incorrect result. The test should render instead `Ed.`. (I used Zotero to compare). The proposed test-data correction makes the test future-safe.

Notice that I have another issue yet to analyze and report: there are inconsistencies between citeproc-java and citeproc-js rendering some entries with non-expected bib fields. I would expect this test breaking again if this is fixed.
